Reference/Common-Pitfalls/index-v9.md

```diff
@@ -0,0 +1,426 @@
+
+So... in the above example, the `UmbracoHelper` which has a request scoped lifetime, will now be statically assigned to a variable, which means that this particular request scoped object is now bound to an Application scope lifetime and will never go away. This could mean that under certain circumstances that an entire Umbraco
+cache copy is stuck in memory, or that the `Security` property of the context will be accessed by multiple threads but this now contains the security information for a user for another request!
+
+Additionally since V9 comes with depdency injection out of the box, there's never really any reason to use static references, instead you should always use inject your required resources, and let the DI container handle the lifetimes of the objects.
+
+## Querying with Descendants, DescendantsOrSelf
+
+It's not 100% bad that you use these queries, you need to understand the implications.
+Here's a particularly bad scenario:
+
+You have 10,000 content items in your tree and your tree structure is something like this:
+
+    - Root
+    -- Home
+    -- Blog (list view with 9495 nodes)
+    -- Office Locations (list view with 500 nodes)
+    -- About Us
+    -- Contact Us
+
+You create a menu on your Home page like:
+
+```csharp
+<ul>
+	<li><a href="@Model.Root().Url()">@Model.Root().Name</a></li>
+	@foreach (var node in Model.Root().DescendantsOrSelf().Where(x => x.Level == 2))
+	{
+		<li><a href="@node.Url()">@node.Name</a></li>
+	}
+</ul>
+```
+
+Which renders out: _Root, Home, Blog, Office Locations, About Us, Contact Us_
+
+BUT!...  this is going to perform most horribly. This is going to iterate over every single node in Umbraco, all 10,000 of them. Further more,
+this means it is going to allocate 10,000 `IPublishedContent` instances in memory in order to check its `Level` value.
+
+This can be re-written as:
+
+```csharp
+<ul>
+	<li><a href="@Model.Root().Url()">@Model.Root().Name</a></li>
+	@foreach (var node in Model.Root().Children)
+	{
+		<li><a href="@node.Url()">@node.Name</a></li>
+	}
+</ul>
+```
+
+In many cases you might know that there is only ever going to be a small number of Descendants . If so then go nuts and use Descendants or DescendantsOrSelf. It's important to be aware of the implications of what you are writing.
+
+## Too much querying (Over querying)
+
+Querying content is not Free! Anytime you make a query or resolve a property value be aware that there is overhead involved.
+You could try to think about every query you make as an SQL call - you don't want to make too many otherwise the performance of
+your website is going to suffer.
```

Comment:
```suggestion
In many cases, you might know that there is only ever going to be a small number of Descendants. If so then go nuts and use Descendants or DescendantsOrSelf. It's important to be aware of the implications of what you are writing.

## Too much querying (Over querying)

Querying content is not Free! Anytime you make a query or resolve a property value be aware that there is overhead involved.
You could try to think about every query you make as an SQL call - you don't want to make too many otherwise the performance of
your website is going to suffer.
```